promptRemove/management.cattle.io.project.vue

```diff
@@ -0,0 +1,154 @@
+<script>
+import { mapState, mapGetters } from 'vuex';
+import { sortBy } from '@/utils/sort';
+import { PROJECT } from '@/config/labels-annotations';
+import { MANAGEMENT, NAMESPACE } from '@/config/types';
+import Checkbox from '@/components/form/Checkbox';
+
+export default {
+  name: 'ProjectAndNamespacesPromptRemove',
+
+  components: { Checkbox },
+
+  props: {
+    value: {
+      type:    Array,
+      default: () => {
+        return [];
+      }
+    },
+
+    type: {
+      type:     String,
+      required: true
+    }
+  },
+
+  async fetch() {
+    this.allNamespaces = await this.$store.dispatch('cluster/findAll', { type: NAMESPACE });
+    this.allProjects = await this.$store.dispatch('management/findAll', { type: MANAGEMENT.PROJECT });
+  },
+
+  data() {
+    return {
+      deleteProjectNamespaces: false,
+      allNamespaces:           [],
+      allProjects:             [],
+    };
+  },
+
+  computed:   {
+    ...mapState('action-menu', ['toRemove']),
+    ...mapGetters({ t: 'i18n/t' }),
+    ...mapGetters(['currentCluster']),
+
+    parentComponent() {
+      return this.$parent.$parent.$parent;
+    },
+
+    currentProject() {
+      if ( this.allProjects) {
+        return this.allProjects?.filter(p => p.spec.displayName === this.parentComponent.nameToMatch)[0];
+      }
+
+      return {};
+    },
+
+    filteredNamespaces() {
+      if (this.currentProject) {
+        return this.allNamespaces.filter(n => n.metadata.labels[PROJECT] === this.currentProject.metadata?.name);
+      }
+
+      return [];
+    },
+
+    plusMore() {
+      const remaining = this.filteredNamespaces.length > 5 ? this.filteredNamespaces.length - 5 : 0;
+
+      return this.t('promptRemove.andOthers', { count: remaining });
+    },
+
+    displayName() {
+      return this.currentProject?.spec?.displayName;
+    },
+
+    names() {
+      return this.filteredNamespaces.map(obj => obj.nameDisplay).slice(0, 5);
+    },
+
+    formattedNamespaces() {
+      return this.filteredNamespaces.reduce((res, namespace, i) => {
+        if (i >= 5) {
+          return res;
+        }
+
+        res += `<b>${ namespace.name }</b>`;
+
+        if (i === this.names.length - 1) {
+          res += this.plusMore;
+        } else {
+          res += i === this.filteredNamespaces.length - 2 ? ' and ' : ', ';
```

Comment:
Need to think about localisation - not all locales will concatenate like this. This duplicates similar code from `promptRemove` - would be better to refactor that into a utility function for formatting a set resource names like this.